lms/djangoapps/courseware/tests/test_credit_requirements.py

```diff
@@ -133,6 +135,23 @@ def test_credit_requirements_not_eligible(self):
         )
         self.assertContains(response, "Verification Failed")
 
+    @ddt.data('honor', 'audit', 'verified', 'credit')
+    def test_credit_requirements_non_credit_enrollment(self, enrollment_mode):
+        # Test the progress table is only displayed to the
+        # verified and credit students.
+        self.enrollment.mode = enrollment_mode
+        self.enrollment.save()  # pylint: disable=no-member
+
+        # Check the progress page display
+        response = self._get_progress_page()
+        if enrollment_mode in ['credit', 'verified']:
```

Comment:
At this point, you should write two tests, one for credit and verified, and one for honor and audit. Also, instead of specifying the strings here, use the class variables set on the `CourseMode` model: https://github.com/edx/edx-platform/blob/master/common/djangoapps/course_modes/models.py#L96-L102